cli/bptest/convert.go

```diff
@@ -0,0 +1,281 @@
+	// write discover test
+	err = writeFile(path.Join(intTestPath, "discover_test.go"), discoverTest)
+	if err != nil {
+		return fmt.Errorf("error writing discover_test.go: %v", err)
+	}
+	testDirs, err := getCurrentTestDirs()
+	if err != nil {
+		return fmt.Errorf("error getting current test dirs: %v", err)
+	}
+	for _, dir := range testDirs {
+		err = convertTest(path.Join(intTestPath, dir))
+		if err != nil {
+			return fmt.Errorf("error converting %s: %v", dir, err)
+		}
+	}
+	// remove kitchen
+	err = os.Remove(".kitchen.yml")
+	if err != nil {
+		return fmt.Errorf("error removing .kitchen.yml: %v", err)
+	}
+	// convert build file
+	// We use build to identify commands to update and update the commands in the buildFile.
+	// This minimizes unnecessary diffs in build yaml due to round tripping.
+	build, buildFile, err := getBuildFromFile(intTestBuildFilePath)
+	if err != nil {
+		return fmt.Errorf("error unmarshalling %s: %v", intTestBuildFilePath, err)
+	}
+	newBuildFile, err := transformBuild(build, buildFile)
+	if err != nil {
+		return fmt.Errorf("error transforming buildfile: %v", err)
+	}
+	return writeFile(intTestBuildFilePath, newBuildFile)
+}
+
+// getCurrentTestDirs returns current test dirs in intTestPath
+func getCurrentTestDirs() ([]string, error) {
+	files, err := ioutil.ReadDir(intTestPath)
+	if err != nil {
+		return nil, err
+	}
+	var dirs []string
+	for _, f := range files {
+		if f.IsDir() {
+			dirs = append(dirs, f.Name())
+		}
+	}
+	return dirs, nil
+}
+
+// convertTest converts a kitchen test in dir to blueprint test
+func convertTest(dir string) error {
+	// read inspec.yaml
+	f, err := ioutil.ReadFile(path.Join(dir, inspecInputsFile))
+	if err != nil {
+		return fmt.Errorf("error reading inspec file: %s", err)
+	}
+	var inspec inspecInputs
+	err = yaml.Unmarshal(f, &inspec)
+	if err != nil {
+		return fmt.Errorf("error unmarshalling inspec file: %s", err)
+	}
+	// get inspec input attributes
+	var inputs []string
+	for _, i := range inspec.Attributes {
+		inputs = append(inputs, i.Name)
+	}
+	// get bpt skeleton
+	testName := path.Base(dir)
+	bpTest, err := getBPTestFromTmpl(testName, inputs)
+	if err != nil {
+		return fmt.Errorf("error creating blueprint test: %s", err)
+	}
+	// remove old test
+	err = os.RemoveAll(dir)
+	if err != nil {
+		return fmt.Errorf("error removing old test dir: %s", err)
+	}
+	// write bpt
+	err = os.MkdirAll(dir, os.ModePerm)
+	if err != nil {
+		return fmt.Errorf("error creating test dir: %s", err)
+	}
+	return writeFile(path.Join(dir, fmt.Sprintf("%s_test.go", strcase.ToSnake(testName))), bpTest)
+}
+
+// getTestFnName returns the go test function name
+func getTestFnName(name string) string {
+	return fmt.Sprintf("Test%s", strcase.ToCamel(name))
+}
+
+// getBPTestFromTmpl returns a skeleton blueprint test
+func getBPTestFromTmpl(testName string, inputs []string) (string, error) {
+	pkgName := strcase.ToSnake(testName)
+	fnName := getTestFnName(testName)
+	tmpl := `package {{.PkgName}}
```

Comment:
Since we've got a lot of templates intermingled throughout this file, I think it would be helpful to externalize them into their own files.

Reply:
Good idea, done